modules/ethereum/src/mock.rs

```diff
@@ -0,0 +1,168 @@
+	Perbill,
+};
+
+pub type AccountId = u64;
+
+#[derive(Clone, Eq, PartialEq)]
+pub struct TestRuntime;
+
+impl_outer_origin! {
+	pub enum Origin for TestRuntime where system = frame_system {}
+}
+
+parameter_types! {
+	pub const BlockHashCount: u64 = 250;
+	pub const MaximumBlockWeight: Weight = 1024;
+	pub const MaximumBlockLength: u32 = 2 * 1024;
+	pub const AvailableBlockRatio: Perbill = Perbill::one();
+}
+
+impl frame_system::Trait for TestRuntime {
+	type Origin = Origin;
+	type Index = u64;
+	type Call = ();
+	type BlockNumber = u64;
+	type Hash = H256;
+	type Hashing = BlakeTwo256;
+	type AccountId = AccountId;
+	type Lookup = IdentityLookup<Self::AccountId>;
+	type Header = SubstrateHeader;
+	type Event = ();
+	type BlockHashCount = BlockHashCount;
+	type MaximumBlockWeight = MaximumBlockWeight;
+	type DbWeight = ();
+	type BlockExecutionWeight = ();
+	type ExtrinsicBaseWeight = ();
+	type AvailableBlockRatio = AvailableBlockRatio;
+	type MaximumBlockLength = MaximumBlockLength;
+	type Version = ();
+	type ModuleToIndex = ();
+	type AccountData = ();
+	type OnNewAccount = ();
+	type OnKilledAccount = ();
+}
+
+parameter_types! {
+	pub const TestAuraConfiguration: AuraConfiguration = test_aura_config();
+	pub const TestValidatorsConfiguration: ValidatorsConfiguration = test_validators_config();
+}
+
+impl Trait for TestRuntime {
+	type AuraConfiguration = TestAuraConfiguration;
+	type ValidatorsConfiguration = TestValidatorsConfiguration;
+	type OnHeadersSubmitted = ();
+}
+
+/// Aura configuration that is used in tests by default.
+pub fn test_aura_config() -> AuraConfiguration {
+	AuraConfiguration {
+		empty_steps_transition: u64::max_value(),
+		strict_empty_steps_transition: 0,
+		validate_step_transition: 0x16e360,
+		validate_score_transition: 0x41a3c4,
+		two_thirds_majority_transition: u64::max_value(),
+		min_gas_limit: 0x1388.into(),
+		max_gas_limit: U256::max_value(),
+		maximum_extra_data_size: 0x20,
+	}
+}
+
+/// Validators configuration that is used in tests by default.
+pub fn test_validators_config() -> ValidatorsConfiguration {
+	ValidatorsConfiguration::Single(ValidatorsSource::List(validators_addresses(3)))
+}
+
+/// Genesis header that is used in tests by default.
+pub fn genesis() -> Header {
+	Header {
+		seal: vec![vec![42].into(), vec![].into()],
+		..Default::default()
+	}
+}
+
+/// Build default i-th block, using data from runtime storage.
+pub fn block_i(number: u64, validators: &[KeyPair]) -> Header {
+	custom_block_i(number, validators, |_| {})
+}
+
+/// Build custom i-th block, using data from runtime storage.
+pub fn custom_block_i(number: u64, validators: &[KeyPair], customize: impl FnOnce(&mut Header)) -> Header {
+	let validator_index: u8 = (number % (validators.len() as u64)) as _;
+	let mut header = Header {
+		number,
+		parent_hash: HeadersByNumber::get(number - 1).unwrap()[0].clone(),
+		gas_limit: 0x2000.into(),
+		author: validator(validator_index).address().to_fixed_bytes().into(),
```

Comment:
I've seen this a few times, could probably be put behind a function so we can just do `author: author(i)`